server/layout_test.go

```diff
@@ -1,79 +1,183 @@
+			},
+			[]chronograf.Layout{
+				chronograf.Layout{
+					ID:          "d20a21c8-69f1-4780-90fe-e69f5e4d138c",
+					Application: "influxdb",
+					Measurement: "influxdb",
+					Cells: []chronograf.Cell{
+						{
+							X:    0,
+							Y:    0,
+							W:    4,
+							H:    4,
+							I:    "3b0e646b-2ca3-4df2-95a5-fd80915459dd",
+							Name: "A Graph",
+						},
 					},
 				},
 			},
-			ExpectedID:     0,
-			ExpectedHref:   "/chronograf/v1/layouts/0",
-			ExpectedStatus: http.StatusCreated,
+			"",
+			false,
 		},
 	}
 
-	for _, test := range tests {
-		// The mocked backing store will be used to
-		// check stored values.
-		store := server.Store{
-			LayoutStore: &mock.LayoutStore{
-				AddError: test.AddError,
-				Layouts:  test.ExistingLayouts,
-			},
-		}
+	for _, test := range layoutTests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			// setup mock chronograf.Service and mock logger
+			lg := &mocks.TestLogger{}
+			svc := server.Service{
+				LayoutStore: &mocks.LayoutStore{
+					AllF: func(ctx context.Context) ([]chronograf.Layout, error) {
+						if len(test.allLayouts) == 0 {
+							return []chronograf.Layout{
+								test.expected,
+							}, nil
+						} else {
+							return test.allLayouts, nil
+						}
+					},
+				},
+				Logger: lg,
+			}
+
+			// setup mock request and response
+			rr := httptest.NewRecorder()
+			reqURL := url.URL{
+				Path: "/chronograf/v1/layouts",
+			}
+			params := reqURL.Query()
+
+			// add query params required by test
+			if test.focusedApp != "" {
+				params.Add("app", test.focusedApp)
+			}
+
+			// re-inject query params
+			reqURL.RawQuery = params.Encode()
+
+			req := httptest.NewRequest("GET", reqURL.RequestURI(), strings.NewReader(""))
+
+			// invoke handler for layouts endpoint
+			svc.Layouts(rr, req)
+
+			// create a throwaway frame to unwrap Layouts
+			respFrame := struct {
+				Layouts []struct {
+					chronograf.Layout
+					Link interface{} `json:"-"`
+				} `json:"layouts"`
+			}{}
+
+			// decode resp into respFrame
+			resp := rr.Result()
+			if err := json.NewDecoder(resp.Body).Decode(&respFrame); err != nil {
+				t.Fatalf("%q - Error unmarshaling JSON: err: %s", test.name, err.Error())
+			}
 
-		// Send the test layout to the mocked store.
-		params := op.PostLayoutsParams{
-			Layout: test.NewLayout,
-		}
-		resp := store.NewLayout(context.Background(), params)
-		w := httptest.NewRecorder()
-		resp.WriteResponse(w, runtime.JSONProducer())
-		if w.Code != test.ExpectedStatus {
-			t.Fatalf("Expected status %d; actual %d", test.ExpectedStatus, w.Code)
-		}
-		loc := w.Header().Get("Location")
-		if loc != test.ExpectedHref {
-			t.Fatalf("Expected status %s; actual %s", test.ExpectedHref, loc)
-		}
+			// compare actual and expected
+			if !cmp.Equal(test.expected, respFrame.Layouts[0].Layout) {
+				t.Fatalf("%q - Expected layouts to be equal: diff:\n\t%s", test.name, cmp.Diff(test.expected, respFrame.Layouts[0].Layout))
```

Comment:
I would agree that testing individual properties on a lower-level test is better. This is more of a mid-level test since we're interested in the output of the service. The shape is important to assert so that we can provide guarantees to the frontend. If the shape of `Axis` changes in a way that violates this test's expectations (e.g. `Bounds` is renamed to `Extents`) this will immediately not compile and shine a light on the fact that you may have broken some frontend expectations.